tests/integration/targets/test_azure_virtual_machine_with_public_ip/tasks/main.yml

```
@@ -21,78 +25,168 @@
           subnet_address_prefixes_cidr:
           - "10.1.0.0/24"
 
-    # Test: Create Virtual machine with default options
-    - name: Create Virtual machine with default network interface
+    # Create VM with auto-created resorces within a new resource group.
+    # Stop, deallocate, start and restart VM
+    - name: Perform valid operations with VM within a new resource group
+      ansible.builtin.include_tasks: "azure_vm_tests.yml"
+      vars:
+        create_asserts_list:
+          - vm_info.vms[0].name == azure_virtual_machine_with_public_ip_vm.name
+          - vm_info.vms[0].network_interface_names[0] == azure_virtual_machine_with_public_ip_vm.name
```

```suggestion
          - vm_info.vms[0].network_interface_names[0] == azure_virtual_machine_with_public_ip_vm.network_interfaces
```
I saw this is different from the validation in the second time, which one is true?

---

In the first test - NIC created automatically with VM name and in second test NIC created before VM creation and attached to the VM after